v2/internal/controllers/crd_dataprotection_backupvault_test.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	// The testing package is imported for testing-related functionality.
+	"testing"
+
+	// The gomega package is used for assertions and expectations in tests.
+	. "github.com/onsi/gomega"
+
+	// The dataprotection package contains types and functions related to dataprotection resources.
+	dataprotection "github.com/Azure/azure-service-operator/v2/api/dataprotection/v1api20230101"
+	// The testcommon package includes common testing utilities.
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	// The to package includes utilities for converting values to pointers.
+	"github.com/Azure/azure-service-operator/v2/internal/util/to"
+)
+
+func newBackupVault(tc *testcommon.KubePerTestContext, name string) *dataprotection.BackupVault {
+	rg := tc.CreateTestResourceGroupAndWait()
```

Comment:
minor: I might pull this out and make each test call it explicitly, and then pass the `rg` here as a parameter.

That way other parts of the test have access to the `rg` if they need it, rather than it being sorta hidden in here. That is also the pattern that most of the other `newXyz` test helpers follow AFAIK.

Reply:
Rectified code to send rg as a parameter in the new backup vault function

Reply:
Also, optimized it by creating a separate function for creating a newBackupVault and reusing the BackupPolicy test file.

Similarly, refactored the code for creating a newBackupPolicy so as to reuse it in the backup instance test file.

Thanks for the suggestion!